cloudinit/cmd/devel/hotplug_hook.py

```diff
@@ -237,6 +242,22 @@ def handle_hotplug(hotplug_init: Init, devpath, subsystem, udevaction):
         raise last_exception
 
 
+def enable_hotplug(hotplug_init: Init, subsystem):
+    LOG.debug("Fetching datasource")
+    datasource = hotplug_init.fetch(existing="trust")
+    if not datasource:
+        return
+    if not datasource.get_supported_events([EventType.HOTPLUG]):
+        LOG.debug("hotplug not supported for event of type %s", subsystem)
+        return
+
+    util.ensure_file(stages.HOTPLUG_ENABLED_FILE, preserve_mode=True)
```

Comment:
Would we want this also to be a file check to avoid re-entrant behavior of hotplug is already enabled to avoid trying to write the udev rules file again?

Reply:
Added the event scope to the file as json (for future extension) and only writing if not present, avoiding reentrant overhead / issues. Example:

```sh
$ cat /etc/cloud/hotplug.enabled
{"scopes": ["network"]}
```